tt_eager/tt_dnn/op_library/sdpa/kernels/compute/sdpa_flash_decode.cpp

```diff
@@ -0,0 +1,706 @@
+
+    mul_tiles_init();
+    cb_wait_front(in0_cb, num_tiles);
+    cb_wait_front(in1_cb, num_tiles);
+    for (uint32_t i = 0; i < num_tiles; i++) {
+        acquire_dst(tt::DstMode::Half);
+        mul_tiles(in0_cb, in1_cb, 0, i, 0);
+        cb_pop_front(in0_cb, 1);
+        cb_reserve_back(in0_cb, 1);
+        pack_tile(0, in0_cb);
+        cb_push_back(in0_cb, 1);
+        release_dst(tt::DstMode::Half);
+    }
+    UNPACK(tensix_sync());
+}
+
+void sub_exp_block(uint32_t in0_cb, uint32_t in1_cb, uint32_t out_cb, uint32_t num_tiles) {
+    // Precondition: in0_cb and in1_cb have num_tiles produced
+    // Postcondition: out_cb has num_tiles produced
+    // Postcondition: in0_cb and in1_cb has num_tiles produced
+    // DEBUG_STATUS("XCAE");
+    sub_tiles_init();
+    // DEBUG_STATUS("XCAF");
+    exp_tile_init<true>();
+    // DEBUG_STATUS("XCAG");
+    // DPRINT_MATH(DPRINT << "[C] R ckpt 2.1.1.0 math" << ENDL());
+    // DPRINT_PACK(DPRINT << "[C] R ckpt 2.1.1.0 pack" << ENDL());
+    // DPRINT_UNPACK(DPRINT << "[C] R ckpt 2.1.1.0 unpack" << ENDL());
+    cb_wait_front(in0_cb, num_tiles);
+    // DEBUG_STATUS("XCAH");
+    // DPRINT_MATH(DPRINT << "[C] R ckpt 2.1.1.1 math" << ENDL());
+    // DPRINT_PACK(DPRINT << "[C] R ckpt 2.1.1.1 pack" << ENDL());
+    // DPRINT_UNPACK(DPRINT << "[C] R ckpt 2.1.1.1 unpack" << ENDL());
+    cb_wait_front(in1_cb, num_tiles);
+    // DEBUG_STATUS("XCAI");
+    cb_reserve_back(out_cb, num_tiles);
+    // DEBUG_STATUS("XCAJ");
+
+    // DPRINT << "[C] R ckpt 2.1.1.2" << ENDL();
+
+    for (uint32_t i = 0; i < num_tiles; i++) {
+
+        acquire_dst(tt::DstMode::Half);
+        // DEBUG_STATUS("XCAK");
+
+        sub_tiles(in0_cb, in1_cb, i, i, 0);
+        // DEBUG_STATUS("XCAL");
+
+        exp_tile<true>(0);
+        // DEBUG_STATUS("XCAM");
+
+        pack_tile(0, out_cb);
+        // DEBUG_STATUS("XCAN");
+
+        cb_push_back(out_cb, 1);
+        // DEBUG_STATUS("XCAO");
+        release_dst(tt::DstMode::Half);
+        // DEBUG_STATUS("XCAP");
+    }
+}
+
+void copy_block(uint32_t in_cb, uint32_t out_cb, uint32_t num_tiles) {
+    // Precondition: in_cb has num_tiles produced
+    // Precondition: out_cb has num_tiles free
+    // Postcondition: in_cb has num_tiles consumed
+    // Postcondition: out_cb has num_tiles produced
+
+    copy_tile_to_dst_init_short(in_cb);
+
+    cb_wait_front(in_cb, num_tiles);
+    cb_reserve_back(out_cb, num_tiles);
+
+    #pragma GCC unroll 0
+    for (uint32_t i = 0; i < num_tiles; i++) {
+        acquire_dst(tt::DstMode::Half);
+        copy_tile(in_cb, i, 0/*dst*/);
+        pack_tile(0, out_cb);
+        cb_push_back(out_cb, 1);
+        release_dst(tt::DstMode::Half);
+    }
+    cb_pop_front(in_cb, num_tiles);
+}
+
+void matmul_blocks(const uint32_t& in0_cb, const uint32_t& in1_cb, const uint32_t& out_cb, const uint32_t& M, const uint32_t& N, const uint32_t& K, const uint32_t& num_blocks, const uint32_t& in0_num_subblocks, const uint32_t& in1_num_subblocks,
```

Comment:
Can you choose a different name? There is `matmul_block` in LLKs and I think it can be confusing.

Reply:
changed